portals/publisher/src/main/webapp/source/src/app/components/Shared/PoliciesUI/AttachedPolicyCard.tsx

```diff
@@ -59,28 +60,30 @@ interface AttachedPolicyCardSharedProps {
     PolicyConfigurationEditDrawer: any;
 }
 
+// Option 1: `listOriginatedFromCommonPolicies` and `isApiRevision` are provided
+interface AttachedPolicyCardWithCommonProps extends AttachedPolicyCardSharedBaseProps {
+    listOriginatedFromCommonPolicies: string[];
+    isApiRevision: boolean;
+}
+
+// Option 2: Neither `listOriginatedFromCommonPolicies` nor `isApiRevision` are provided
+interface AttachedPolicyCardWithoutCommonProps extends AttachedPolicyCardSharedBaseProps {
+    listOriginatedFromCommonPolicies?: undefined;
+    isApiRevision?: undefined;
+}
+
+// Combine the two using a union type
+type AttachedPolicyCardSharedProps = AttachedPolicyCardWithCommonProps | AttachedPolicyCardWithoutCommonProps;
+
 /**
  * Renders a single sortable policy card.
  * @param {any} AttachedPolicyCardProps Input props from parent components.
  * @returns {TSX} Sortable attached policy card UI.
  */
-const AttachedPolicyCardShared: FC<AttachedPolicyCardSharedProps> = ({
-    policyObj,
-    currentFlow,
-    verb,
-    target,
-    allPolicies,
-    isAPILevelPolicy,
-    drawerOpen,
-    handleDrawerOpen,
-    handlePolicyDownload,
-    handleDelete,
-    setDrawerOpen,
-    PolicyConfigurationEditDrawer
-}) => {
+const AttachedPolicyCardShared: FC<AttachedPolicyCardSharedProps> = (props) => {
 
-    const policyColor = Utils.stringToColor(policyObj.displayName);
-    const policyBackgroundColor = drawerOpen
+    const policyColor = Utils.stringToColor(props.policyObj.displayName);
+    const policyBackgroundColor = props.drawerOpen
```

Comment:
Let's not do these modifications. Check and fix similar places as well.

```suggestion
    const policyColor = Utils.stringToColor(policyObj.displayName);
    const policyBackgroundColor = drawerOpen
```

Reply:
relates to https://github.com/wso2/apim-apps/pull/809#discussion_r1901575381